weavertest/init.go

```diff
@@ -142,11 +149,23 @@ func (r Runner) Bench(b *testing.B, testBody any) {
 
 func (r Runner) sub(t testing.TB, isBench bool, testBody any) {
 	t.Helper()
-	runner, err := checkRunFunc(t, testBody)
+	body, intfs, err := checkRunFunc(t, testBody)
 	if err != nil {
 		t.Fatal(fmt.Errorf("weavertest.Run argument: %v", err))
 	}
 
+	// Assume a component Foo with implementing struct foo. We disallow tests
+	// like the one below where the user provides a fake and a component
+	// implementation pointer for the same component.
```

Comment:
Why do we need to disallow this?

Reply:
We could allow it! In code like the following, the fake is unused because of the presence of the `*foo` argument. I thought it was better to fail the test rather than letting the fake go silently unused? What do you think? Happy to allow it.

```go
runner.Fakes = append(runner.Fakes, weavertest.Fake[Foo](...))
runner.Test(t, func(t *testing.T, impl *foo) {...})
```